frontend/public/components/dashboard/project-dashboard/project-dashboard.tsx

```diff
@@ -26,13 +26,29 @@ const getNamespaceDashboardConsoleLinks = (
   ns: K8sResourceKind,
   consoleLinks: K8sResourceKind[],
 ): K8sResourceKind[] => {
-  return _.filter(consoleLinks, (link: K8sResourceKind) => {
-    if (link.spec.location !== 'NamespaceDashboard') {
-      return false;
-    }
-    const namespaces: string[] = _.get(link, ['spec', 'namespaceDashboard', 'namespaces']);
-    return _.isEmpty(namespaces) || _.includes(namespaces, ns.metadata.name);
-  });
+  return _.filter(
+    consoleLinks,
+    (link: K8sResourceKind): boolean => {
+      if (link.spec.location !== 'NamespaceDashboard') {
+        return false;
+      }
+
+      const namespaces: string[] = _.get(link, 'spec.namespaceDashboard.namespaces');
+      const selector: Selector = _.get(link, 'spec.namespaceDashboard.namespaceSelector');
+
+      // If neither namespaces or selector was provided, show the link for all namespaces.
+      if (_.isEmpty(namespaces) && _.isEmpty(selector)) {
+        return true;
+      }
+
+      // Show the link if either namespaces or the selector matches this namespace.
```

Comment:
Comment is incorret?

Reply:
I think it's right? We show the link if we match either namespaces OR selector. (Maybe I should rephrase.)

Reply:
The comment is correct but does not reflect the code change, but overall logic.
Check https://github.com/openshift/api/blob/master/console/v1/types_console_link.go#L48

Reply:
Isn't the API doc saying the same thing?